token/core/zkatdlog/crypto/setup.go

```diff
@@ -193,19 +204,24 @@ func (pp *PublicParams) GeneratePedersenParameters() error {
 	return nil
 }
 
-func (pp *PublicParams) GenerateRangeProofParameters(signer *pssign.Signer, maxValue uint) error {
+func (pp *PublicParams) GenerateRangeProofParameters(bitLength int) error {
+	if bitLength <= 0 {
+		return errors.Errorf("invalid bit length, must be larger than 0")
+	}
 	curve := mathlib.Curves[pp.Curve]
-	pp.RangeProofParams = &RangeProofParams{Q: signer.Q, SignPK: signer.PK}
-
-	pp.RangeProofParams.SignedValues = make([]*pssign.Signature, maxValue)
-	for i := 0; i < len(pp.RangeProofParams.SignedValues); i++ {
-		var err error
-		m := make([]*mathlib.Zr, 1)
-		m[0] = curve.NewZrFromInt(int64(i))
-		pp.RangeProofParams.SignedValues[i], err = signer.Sign(m)
-		if err != nil {
-			return errors.Errorf("failed to generate public parameters: cannot sign range")
-		}
+
+	pp.RangeProofParams = &RangeProofParams{
+		P:              curve.HashToG1([]byte(strconv.Itoa(0))),
+		Q:              curve.HashToG1([]byte(strconv.Itoa(1))),
+		BitLength:      bitLength,
+		NumberOfRounds: int(math.Log2(float64(bitLength))),
+	}
+	pp.RangeProofParams.LeftGenerators = make([]*mathlib.G1, bitLength)
+	pp.RangeProofParams.RightGenerators = make([]*mathlib.G1, bitLength)
+
+	for i := 0; i < bitLength; i++ {
+		pp.RangeProofParams.LeftGenerators[i] = curve.HashToG1([]byte(strconv.Itoa(2 * (i + 1))))
```

Comment:
Maybe it's worth to add some prefix "range proof" to the hash's input, otherwise in other protocols they would be the same generators?

Reply:
OK.